# Домашнє завдання до теми "Основи роботи з даними у Pandas"

В цьому домашньому завданні продовжуємо працювати з набором даних `supermarket_sales.csv`.

0. Імпортуйте бібліотеку pandas та зчитайте дані у змінну `df` типу `pandas.DataFrame`.

In [1]:
import pandas as pd

In [247]:
df = pd.read_csv('/Users/s1ndly/Data Analytics/data/supermarket_sales.csv')

In [26]:
df[0:20]

Invoice ID Branch       City Customer type  Gender  \
0   750-67-8428      A     Yangon        Member  Female   
1   226-31-3081      C  Naypyitaw        Normal  Female   
2   631-41-3108      A     Yangon        Normal    Male   
3   123-19-1176      A     Yangon        Member    Male   
4   373-73-7910      A     Yangon        Normal    Male   
5   699-14-3026      C  Naypyitaw        Normal    Male   
6   355-53-5943      A     Yangon        Member  Female   
7   315-22-5665      C  Naypyitaw        Normal  Female   
8   665-32-9167      A     Yangon        Member  Female   
9   692-92-5582      B   Mandalay        Member  Female   
10  351-62-0822      B   Mandalay        Member  Female   
11  529-56-3974      B   Mandalay        Member    Male   
12  365-64-0515      A     Yangon        Normal  Female   
13  252-56-2699      A     Yangon        Normal    Male   
14  829-34-3910      A     Yangon        Normal  Female   
15  299-46-1805      B   Mandalay        Member  Female   
16  656-95-9349      A     Yangon        Member  Female   
17  765-26-6951      A     Yangon        Normal    Male   
18  329-62-1586      A     Yangon        Normal    Male   
19  319-50-3348      B   Mandalay        Normal  Female   

              Product line  Unit price  Quantity   Tax 5%     Total  \
0        Health and beauty       74.69         7  26.1415  548.9715   
1   Electronic accessories       15.28         5   3.8200   80.2200   
2       Home and lifestyle       46.33         7  16.2155  340.5255   
3        Health and beauty       58.22         8  23.2880  489.0480   
4        Sports and travel       86.31         7  30.2085  634.3785   
5   Electronic accessories       85.39         7  29.8865  627.6165   
6   Electronic accessories       68.84         6  20.6520  433.6920   
7       Home and lifestyle       73.56        10  36.7800  772.3800   
8        Health and beauty       36.26         2   3.6260   76.1460   
9       Food and beverages       54.84         3   8.2260  172.7460   
10     Fashion accessories       14.48         4   2.8960   60.8160   
11  Electronic accessories       25.51         4   5.1020  107.1420   
12  Electronic accessories       46.95         5  11.7375  246.4875   
13      Food and beverages       43.19        10  21.5950  453.4950   
14       Health and beauty       71.38        10  35.6900  749.4900   
15       Sports and travel       93.72         6  28.1160  590.4360   
16       Health and beauty       68.93         7  24.1255  506.6355   
17       Sports and travel       72.61         6  21.7830  457.4430   
18      Food and beverages       54.67         3   8.2005  172.2105   
19      Home and lifestyle       40.30         2   4.0300   84.6300   

         Date   Time      Payment    cogs  gross margin percentage  \
0    1/5/2019  13:08      Ewallet  522.83                 4.761905   
1    3/8/2019  10:29         Cash   76.40                 4.761905   
2    3/3/2019  13:23  Credit card  324.31                 4.761905   
3   1/27/2019  20:33      Ewallet  465.76                 4.761905   
4    2/8/2019  10:37      Ewallet  604.17                 4.761905   
5   3/25/2019  18:30      Ewallet  597.73                 4.761905   
6   2/25/2019  14:36      Ewallet  413.04                 4.761905   
7   2/24/2019  11:38      Ewallet  735.60                 4.761905   
8   1/10/2019  17:15  Credit card   72.52                 4.761905   
9   2/20/2019  13:27  Credit card  164.52                 4.761905   
10   2/6/2019  18:07      Ewallet   57.92                 4.761905   
11   3/9/2019  17:03         Cash  102.04                 4.761905   
12  2/12/2019  10:25      Ewallet  234.75                 4.761905   
13   2/7/2019  16:48      Ewallet  431.90                 4.761905   
14  3/29/2019  19:21         Cash  713.80                 4.761905   
15  1/15/2019  16:19         Cash  562.32                 4.761905   
16  3/11/2019  11:03  Credit card  482.51                 4.761905   
17   1/1/2019

1. За допомогою бібліотеки Pandas виведіть перші 4 і останні 4 записи у наборі даних (це дві окремі команди). Треба саме 4! Не 5.

In [7]:
df.head(4)

Invoice ID Branch       City Customer type  Gender  \
0  750-67-8428      A     Yangon        Member  Female   
1  226-31-3081      C  Naypyitaw        Normal  Female   
2  631-41-3108      A     Yangon        Normal    Male   
3  123-19-1176      A     Yangon        Member    Male   

             Product line  Unit price  Quantity   Tax 5%     Total       Date  \
0       Health and beauty       74.69         7  26.1415  548.9715   1/5/2019   
1  Electronic accessories       15.28         5   3.8200   80.2200   3/8/2019   
2      Home and lifestyle       46.33         7  16.2155  340.5255   3/3/2019   
3       Health and beauty       58.22         8  23.2880  489.0480  1/27/2019   

    Time      Payment    cogs  gross margin percentage  gross income  Rating  
0  13:08      Ewallet  522.83                 4.761905       26.1415     9.1  
1  10:29         Cash   76.40                 4.761905        3.8200     9.6  
2  13:23  Credit card  324.31                 4.761905       16.2155     7.4  
3  20:33      Ewallet  465.76                 4.761905       23.2880     8.4

In [8]:
df.tail(4)

Invoice ID Branch      City Customer type  Gender         Product line  \
996  303-96-2227      B  Mandalay        Normal  Female   Home and lifestyle   
997  727-02-1313      A    Yangon        Member    Male   Food and beverages   
998  347-56-2442      A    Yangon        Normal    Male   Home and lifestyle   
999  849-09-3807      A    Yangon        Member  Female  Fashion accessories   

     Unit price  Quantity  Tax 5%     Total       Date   Time  Payment  \
996       97.38        10  48.690  1022.490   3/2/2019  17:16  Ewallet   
997       31.84         1   1.592    33.432   2/9/2019  13:22     Cash   
998       65.82         1   3.291    69.111  2/22/2019  15:33     Cash   
999       88.34         7  30.919   649.299  2/18/2019  13:28     Cash   

       cogs  gross margin percentage  gross income  Rating  
996  973.80                 4.761905        48.690     4.4  
997   31.84                 4.761905         1.592     7.7  
998   65.82                 4.761905         3.291     4.1  
999  618.38                 4.761905        30.919     6.6

Очікуваний результат:

![](https://drive.google.com/uc?export=view&id=1F5XlzBOOTMsedpsYqz6NnXjIPTDe6jpx)

2. Виведіть рядки даних з індексами з 615 по 619 включно.

In [10]:
df[615:620]

Invoice ID Branch       City Customer type  Gender  \
615  509-29-3912      A     Yangon        Member  Female   
616  327-40-9673      B   Mandalay        Member    Male   
617  840-19-2096      C  Naypyitaw        Member    Male   
618  828-46-6863      A     Yangon        Member    Male   
619  641-96-3695      C  Naypyitaw        Member  Female   

               Product line  Unit price  Quantity   Tax 5%     Total  \
615       Sports and travel       38.72         9  17.4240  365.9040   
616       Sports and travel       72.60         6  21.7800  457.3800   
617  Electronic accessories       87.91         5  21.9775  461.5275   
618      Food and beverages       98.53         6  29.5590  620.7390   
619     Fashion accessories       43.46         6  13.0380  273.7980   

          Date   Time      Payment    cogs  gross margin percentage  \
615  3/20/2019  12:24      Ewallet  348.48                 4.761905   
616  1/13/2019  19:51         Cash  435.60                 4.761905   
617  3/14/2019  18:10      Ewallet  439.55                 4.761905   
618  1/23/2019  11:22  Credit card  591.18                 4.761905   
619   2/7/2019  17:55      Ewallet  260.76                 4.761905   

     gross income  Rating  
615       17.4240     4.2  
616       21.7800     6.9  
617       21.9775     4.4  
618       29.5590     4.0  
619       13.0380     8.5

Очікуваний результат:

![](https://drive.google.com/uc?export=view&id=1Otufe_TXRmjZxWigzBiVBLCAVUfEReXM)

3. Виведіть назви колонок у наборі даних разом з їхніми типами. Це можна зробити кількома способами, важливо аби в результаті були назви колонок і відповідні їм типи.

In [11]:
df.dtypes

Invoice ID                  object
Branch                      object
City                        object
Customer type               object
Gender                      object
Product line                object
Unit price                 float64
Quantity                     int64
Tax 5%                     float64
Total                      float64
Date                        object
Time                        object
Payment                     object
cogs                       float64
gross margin percentage    float64
gross income               float64
Rating                     float64
dtype: object

In [13]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 1000 entries, 0 to 999
Data columns (total 17 columns):
 #   Column                   Non-Null Count  Dtype  
---  ------                   --------------  -----  
 0   Invoice ID               1000 non-null   object 
 1   Branch                   1000 non-null   object 
 2   City                     1000 non-null   object 
 3   Customer type            1000 non-null   object 
 4   Gender                   1000 non-null   object 
 5   Product line             1000 non-null   object 
 6   Unit price               1000 non-null   float64
 7   Quantity                 1000 non-null   int64  
 8   Tax 5%                   1000 non-null   float64
 9   Total                    1000 non-null   float64
 10  Date                     1000 non-null   object 
 11  Time                     1000 non-null   object 
 12  Payment                  1000 non-null   object 
 13  cogs                     1000 non-null   float64
 14  gross margin percentage  

Очікуваний результат (один з можливих форматів):

![](https://drive.google.com/uc?export=view&id=1ASh6-S-_f3X9fTQAPIvnHMFZamTBjNkL)

4. Користуючись методом `DataFrame.describe()` виявіть 1, 25, 50, 75 і 99 персентилі колонки `gross income`. Знайдіть, яке значення відповідає 75отму персентилю цієї колонки і що це означає? Як би Ви його пояснили простими словами колегам? Відповідь напишіть текстом в окремій клітинці типу Markdown.

In [21]:
pd.DataFrame(df['gross income'].describe())

gross income
count   1000.000000
mean      15.379369
std       11.708825
min        0.508500
25%        5.924875
50%       12.088000
75%       22.445250
max       49.650000

# This means that the total profitability of supermarkets is $22.445250 or less

or it indicates that the direct profit from sold goods is $22.45 (to 2 dp) or less. THis sugests that shops might be generating insufficient profits (per each purchase), as the remaining 25 percent of the max gross income, ie 22.45$ < 27.20$ (since 49.65-22.45 = 27.20). Therefore, If I were a regional / or top manager at the company, I would recommend to cooperate with the Marketing Departament to look for the items that account for this 75% of gross income, to improve  its financial gains (if it's the main goal).

Depends on the strategy, reliability of data, economic envir


Очікуваний результат:

![](https://drive.google.com/uc?export=view&id=1bDa1YFUWDsPm78Lnpy2geQND8M1HEQiT)

5. Виведіть кількість інвойсів за кожним унікальним містом в наборі даних.

  Підказка: інвойс - це один рядок в цьому наборі даних, ми можемо цим скористатись аби швидко отримати відповідь.

In [32]:
df.City.value_counts()

City
Yangon       340
Mandalay     332
Naypyitaw    328
Name: count, dtype: int64

Очікуваний результат:

![](https://drive.google.com/uc?export=view&id=1G8c6M875NOW7oU3OtZegTOnyD2y8lhAV)

6. Виведіть, який відсоток інвойсів припадає на кожне унікальне місто.

In [40]:
invoice_num = df.City.value_counts()
invoice_num / invoice_num.sum()  #* 100

City
Yangon       34.0
Mandalay     33.2
Naypyitaw    32.8
Name: count, dtype: float64

In [41]:
# OR
df.City.value_counts(normalize=True)


City
Yangon       0.340
Mandalay     0.332
Naypyitaw    0.328
Name: proportion, dtype: float64

Очікуваний результат:

![](https://drive.google.com/uc?export=view&id=1yZZpWhL2y8DX9rSRioYCRh1VbJgcf3fY)

7.1. Використовуючи фільтрацію даних, знайдіть середнє значення `gross income` для міст `Yangon` та `Mandalay`. В якому місті воно більше?

7.2 Сформулюйте 3 гіпотези, чому в одному місті gross income вище, як можна дізнатись чи правдива ця гіпотеза з використанням даних (наявних і додаткових), та як висновки з цих гіпотез можна використати для прийняття рішень як підвищити прибуток в мережі супермаркетів. Відповідь напишіть текстом в окремій клітинці типу Markdown.

Це завдання - на продуктове мислення. Якщо не знаєте, які гіпотези сформулювати - напишіть ті, які здаються вам зовсім примітивними. Важливо навчитись висловлювати свої гіпотези щодо даних, і з часом та досвідом вони будуть ставати щоразу глибше і точніше. Весь аналіз даних будується на тому, що ми перевіряємо певні гіпотези та робимо висновки і рекоемндації на основі своїх досліджень.

In [77]:
city_df = df[['City', 'gross income']]
pd.DataFrame(city_df)

City  gross income
0       Yangon       26.1415
1    Naypyitaw        3.8200
2       Yangon       16.2155
3       Yangon       23.2880
4       Yangon       30.2085
..         ...           ...
995  Naypyitaw        2.0175
996   Mandalay       48.6900
997     Yangon        1.5920
998     Yangon        3.2910
999     Yangon       30.9190

[1000 rows x 2 columns]

In [100]:
city_df2[city_df2.City == 'Yangon'].describe()

Unit price    Quantity      Tax 5%        Total        cogs  \
count  340.000000  340.000000  340.000000   340.000000  340.000000   
mean    54.780853    5.467647   14.874001   312.354031  297.480029   
std     26.132127    2.859876   11.030477   231.640025  220.609547   
min     10.080000    1.000000    0.604500    12.694500   12.090000   
25%     32.250000    3.000000    6.547125   137.489625  130.942500   
50%     53.235000    5.000000   11.468000   240.828000  229.360000   
75%     75.095000    8.000000   21.873375   459.340875  437.467500   
max     99.830000   10.000000   49.490000  1039.290000  989.800000   

       gross margin percentage  gross income      Rating  
count               340.000000    340.000000  340.000000  
mean                  4.761905     14.874001    7.027059  
std                   0.000000     11.030477    1.731345  
min                   4.761905      0.604500    4.000000  
25%                   4.761905      6.547125    5.600000  
50%                   4.761905     11.468000    7.100000  
75%                   4.761905     21.873375    8.500000  
max                   4.761905     49.490000   10.000000

In [101]:
city_df2[city_df2.City == 'Mandalay'].describe()

Unit price    Quantity      Tax 5%        Total        cogs  \
count  332.000000  332.000000  332.000000   332.000000  332.000000   
mean    55.659277    5.481928   15.232024   319.872506  304.640482   
std     26.224559    2.896446   11.545122   242.447565  230.902443   
min     10.590000    1.000000    0.887500    18.637500   17.750000   
25%     33.300000    3.000000    5.711250   119.936250  114.225000   
50%     54.850000    5.000000   12.042000   252.882000  240.840000   
75%     77.002500    8.000000   22.170750   465.585750  443.415000   
max     99.960000   10.000000   48.690000  1022.490000  973.800000   

       gross margin percentage  gross income      Rating  
count               332.000000    332.000000  332.000000  
mean                  4.761905     15.232024    6.818072  
std                   0.000000     11.545122    1.713719  
min                   4.761905      0.887500    4.000000  
25%                   4.761905      5.711250    5.300000  
50%                   4.761905     12.042000    6.700000  
75%                   4.761905     22.170750    8.200000  
max                   4.761905     48.690000   10.000000

In [218]:
city_df2[city_df2.City == 'Mandalay']['Product line'].value_counts()

Product line
Fashion accessories       62
Sports and travel         62
Electronic accessories    55
Health and beauty         53
Food and beverages        50
Home and lifestyle        50
Name: count, dtype: int64

In [219]:
city_df2[city_df2.City == 'Yangon']['Product line'].value_counts()


Product line
Home and lifestyle        65
Electronic accessories    60
Sports and travel         59
Food and beverages        58
Fashion accessories       51
Health and beauty         47
Name: count, dtype: int64

In [242]:
print(f'Yangon - Food and beverages - Mean Price - {city_df2[(city_df2.City == 'Yangon') & (city_df2['Product line'] == 'Food and beverages')]['Unit price'].mean()}')
print(f'Yangon - Food and beverages - SAles volume - {city_df2[(city_df2.City == 'Yangon') & (city_df2['Product line'] == 'Food and beverages')]['Quantity'].mean()}')


print(f'Mandalay - Food and beverages - Mean Price - {city_df2[(city_df2.City == 'Mandalay') & (city_df2['Product line'] == 'Food and beverages')]['Unit price'].mean()}')
print(f'Mandalay - Food and beverages - SAles volume - {city_df2[(city_df2.City == 'Mandalay') & (city_df2['Product line'] == 'Food and beverages')]['Quantity'].mean()}')

Yangon - Food and beverages - Mean Price - 54.97448275862069
Yangon - Food and beverages - SAles volume - 5.396551724137931
Mandalay - Food and beverages - Mean Price - 55.54
Mandalay - Food and beverages - SAles volume - 5.4


In [238]:
print(f'Yangon - Electronic - Mean Price - {city_df2[(city_df2.City == 'Yangon') & (city_df2['Product line'] == 'Electronic accessories')]['Unit price'].mean()}')
print((f'Yangon - Electronic - SAles volume - {city_df2[(city_df2.City == 'Yangon') & (city_df2['Product line'] == 'Electronic accessories')]['Quantity'].mean()}'))


print(f'Mandalay - Electronic - Mean Price - {city_df2[(city_df2.City == 'Mandalay') & (city_df2['Product line'] == 'Electronic accessories')]['Unit price'].mean()}')
print((f'Mandalay - Electronic - SAles volume - {city_df2[(city_df2.City == 'Mandalay') & (city_df2['Product line'] == 'Electronic accessories')]['Quantity'].mean()}'))

Yangon - Electronic - Mean Price - 54.87116666666667
Yangon - Electronic - SAles volume - 5.366666666666666
Mandalay - Electronic - Mean Price - 49.85418181818181
Mandalay - Electronic - SAles volume - 5.745454545454545


# 7.2
Gross Income

Firstly, it my depend on the pricing strategy of the high-priced products ie Electronic Accessories. In Y (Yangon), the average price is $54.87 (to 2 dp), while in M(Mandalay), it is $49.85. However, the mean sales volume in Mandalay is higher than in Yangon. Therefore, to improve profit gains, the price of Electronic Accessories could be increased.

Secondly, the supermarket business could establish new partnerships with Ewallet apps to make it easier for customers to purchase products, as special offers can be found. Since the most poplar type of payment is Ewallet, this could increase sales. (Additionaly, the potential customers could be converted into new members, which would, in turn, generate an extra income stream)

Thirdly, some perishable products, such as Food and Beverages or Health and Beauty items, could generate more income in the short term than Electronic Accessories. As these types of products are more likely to be demanded at a higher rate leading to higher inventory turnover, leading to higher turnover.


In [235]:
print(f'Yangon - Electronic - Mean Price - {city_df2[(city_df2.City == 'Yangon') & (city_df2['Product line'] == 'Electronic accessories')]['Unit price'].mean()}')

Yangon - Electronic - Mean Price - 54.87116666666667


In [210]:
mand_up_mean = city_df2[city_df2.City == 'Mandalay']['Unit price'].mean()
yang_up_mean = city_df2[city_df2.City == 'Yangon']['Unit price'].mean()

mand_tax_mean = city_df2[city_df2.City == 'Mandalay']['Tax 5%'].mean()
yang_tax_mean = city_df2[city_df2.City == 'Yangon']['Tax 5%'].mean()

mand_payment_type = city_df2[city_df2.City == 'Mandalay']['Payment'].value_counts().to_string(index=True, header=False)
yang_payment_type = city_df2[city_df2.City == 'Yangon']['Payment'].value_counts().to_string(index=True, header=False)

mand_sales_volume = city_df2[city_df2.City == 'Mandalay']['Quantity'].mean()
yang_sales_volume = city_df2[city_df2.City == 'Yangon']['Quantity'].mean()

mand_product_line = city_df2[city_df2.City == 'Mandalay']['Product line'].value_counts().to_string(index=True, header=False)
yang_product_line = city_df2[city_df2.City == 'Yangon']['Product line'].value_counts().to_string(index=True, header=False)

pd.set_option('display.max_colwidth', None)


result_df= pd.DataFrame({
    'City': ['Mandalay', 'Yangon'],
    'Unit Price': [mand_up_mean, yang_up_mean],
    'Tax 5%': [mand_tax_mean, yang_tax_mean],
    'Sales Volume': [mand_sales_volume, yang_sales_volume],
    'Payment': [mand_payment_type, yang_payment_type],
    'Product line': [mand_product_line, yang_product_line]
})


display(result_df)

City  Unit Price     Tax 5%  Sales Volume  \
0  Mandalay   55.659277  15.232024      5.481928   
1    Yangon   54.780853  14.874001      5.467647   

                                                      Payment  \
0  Ewallet        113\nCash           110\nCredit card    109   
1  Ewallet        126\nCash           110\nCredit card    104   

                                                                                                                                                                         Product line  
0  Fashion accessories       62\nSports and travel         62\nElectronic accessories    55\nHealth and beauty         53\nFood and beverages        50\nHome and lifestyle        50  
1  Home and lifestyle        65\nElectronic accessories    60\nSports and travel         59\nFood and beverages        58\nFashion accessories       51\nHealth and beauty         47

In [240]:
city_df2[0:50]

Invoice ID Branch      City Customer type  Gender  \
0   750-67-8428      A    Yangon        Member  Female   
2   631-41-3108      A    Yangon        Normal    Male   
3   123-19-1176      A    Yangon        Member    Male   
4   373-73-7910      A    Yangon        Normal    Male   
6   355-53-5943      A    Yangon        Member  Female   
8   665-32-9167      A    Yangon        Member  Female   
9   692-92-5582      B  Mandalay        Member  Female   
10  351-62-0822      B  Mandalay        Member  Female   
11  529-56-3974      B  Mandalay        Member    Male   
12  365-64-0515      A    Yangon        Normal  Female   
13  252-56-2699      A    Yangon        Normal    Male   
14  829-34-3910      A    Yangon        Normal  Female   
15  299-46-1805      B  Mandalay        Member  Female   
16  656-95-9349      A    Yangon        Member  Female   
17  765-26-6951      A    Yangon        Normal    Male   
18  329-62-1586      A    Yangon        Normal    Male   
19  319-50-3348      B  Mandalay        Normal  Female   
21  371-85-5789      B  Mandalay        Normal    Male   
22  273-16-6619      B  Mandalay        Normal    Male   
23  636-48-8204      A    Yangon        Normal    Male   
24  549-59-1358      A    Yangon        Member    Male   
25  227-03-5010      A    Yangon        Member  Female   
26  649-29-6775      B  Mandalay        Normal    Male   
27  189-17-4241      A    Yangon        Normal  Female   
28  145-94-9061      B  Mandalay        Normal  Female   
29  848-62-7243      A    Yangon        Normal    Male   
30  871-79-8483      B  Mandalay        Normal    Male   
31  149-71-6266      B  Mandalay        Member    Male   
32  640-49-2076      B  Mandalay        Normal    Male   
33  595-11-5460      A    Yangon        Normal    Male   
36  129-29-8530      A    Yangon        Member    Male   
37  272-65-1806      A    Yangon        Normal  Female   
39  777-82-7220      B  Mandalay        Member    Male   
40  280-35-5823      B  Mandalay        Member  Female   
42  354-25-5821      B  Mandalay        Member  Female   
45  132-32-9879      B  Mandalay        Member  Female   
46  370-41-7321      B  Mandalay        Member    Male   
47  727-46-3608      B  Mandalay        Member  Female   
48  669-54-1719      B  Mandalay        Member    Male   
51  162-48-8011      A    Yangon        Member  Female   
52  616-24-2851      B  Mandalay        Member  Female   
54  242-55-6721      B  Mandalay        Normal    Male   
56  106-35-6779      A    Yangon        Member    Male   
57  635-40-6220      A    Yangon        Normal    Male   
58  817-48-8732      A    Yangon        Member  Female   
62  347-34-2234      B  Mandalay        Member  Female   
63  199-75-8169      A    Yangon        Member    Male   
64  853-23-2453      B  Mandalay        Member    Male   
65  877-22-3308      A    Yangon        Member    Male   
67  109-28-2512      B  Mandalay        Member  Female   

              Product line  Unit price  Quantity   Tax 5%     Total  \
0        Health and beauty       74.69         7  26.1415  548.9715   
2       Home and lifestyle       46.33         7  16.2155  340.5255   
3        Health and beauty       58.22         8  23.2880  489.0480   
4        Sports and travel       86.31         7  30.2085  634.3785   
6   Electronic accessories       68.84         6  20.6520  433.6920   
8        Health and beauty       36.26         2   3.6260   76.1460   
9       Food and beverages       54.84         3   8.2260  172.7460   
10     Fashion accessories       14.48         4   2.8960   60.8160   
11  Electronic accessories       25.51         4   5.1020  107.1420   
12  Electronic accessories       46.95         5  11.7375  246.4875   
13      Food and beverages       43.19        10  21.5950  453.4950   
14       Health and beauty       71.38        10  35.6900  749.4900   
15       Sports and travel       93.72         6  28.1160  590.4360   
16       Health and beauty       68.93         7  24

# 7.1

In [81]:
city_df2 = df[(df.City == 'Yangon') | (df.City == 'Mandalay')]
# filtered_city = df.query('City == "Yangon" or City == "Mandalay"')
city_df2

Invoice ID Branch      City Customer type  Gender  \
0    750-67-8428      A    Yangon        Member  Female   
2    631-41-3108      A    Yangon        Normal    Male   
3    123-19-1176      A    Yangon        Member    Male   
4    373-73-7910      A    Yangon        Normal    Male   
6    355-53-5943      A    Yangon        Member  Female   
..           ...    ...       ...           ...     ...   
993  690-01-6631      B  Mandalay        Normal    Male   
996  303-96-2227      B  Mandalay        Normal  Female   
997  727-02-1313      A    Yangon        Member    Male   
998  347-56-2442      A    Yangon        Normal    Male   
999  849-09-3807      A    Yangon        Member  Female   

               Product line  Unit price  Quantity   Tax 5%      Total  \
0         Health and beauty       74.69         7  26.1415   548.9715   
2        Home and lifestyle       46.33         7  16.2155   340.5255   
3         Health and beauty       58.22         8  23.2880   489.0480   
4         Sports and travel       86.31         7  30.2085   634.3785   
6    Electronic accessories       68.84         6  20.6520   433.6920   
..                      ...         ...       ...      ...        ...   
993     Fashion accessories       17.49        10   8.7450   183.6450   
996      Home and lifestyle       97.38        10  48.6900  1022.4900   
997      Food and beverages       31.84         1   1.5920    33.4320   
998      Home and lifestyle       65.82         1   3.2910    69.1110   
999     Fashion accessories       88.34         7  30.9190   649.2990   

          Date   Time      Payment    cogs  gross margin percentage  \
0     1/5/2019  13:08      Ewallet  522.83                 4.761905   
2     3/3/2019  13:23  Credit card  324.31                 4.761905   
3    1/27/2019  20:33      Ewallet  465.76                 4.761905   
4     2/8/2019  10:37      Ewallet  604.17                 4.761905   
6    2/25/2019  14:36      Ewallet  413.04                 4.761905   
..         ...    ...          ...     ...                      ...   
993  2/22/2019  18:35      Ewallet  174.90                 4.761905   
996   3/2/2019  17:16      Ewallet  973.80                 4.761905   
997   2/9/2019  13:22         Cash   31.84                 4.761905   
998  2/22/2019  15:33         Cash   65.82                 4.761905   
999  2/18/2019  13:28         Cash  618.38                 4.761905   

     gross income  Rating  
0         26.1415     9.1  
2         16.2155     7.4  
3         23.2880     8.4  
4         30.2085     5.3  
6         20.6520     5.8  
..            ...     ...  
993        8.7450     6.6  
996       48.6900     4.4  
997        1.5920     7.7  
998        3.2910     4.1  
999       30.9190     6.6  

[672 rows x 17 columns]

In [87]:
print(city_df2[city_df2.City == 'Yangon']['gross income'].mean())
print(city_df2[city_df2.City == 'Mandalay']['gross income'].mean())

14.874001470588235
15.232024096385542


In [96]:
Yangon_GI_mean = city_df2.query('City == "Yangon"')['gross income'].mean()
Mandalay_GI_mean = city_df2.query('City == "Mandalay"')['gross income'].mean()
print(Yangon_GI_mean, Mandalay_GI_mean)

14.874001470588235 15.232024096385542


Mandalay > Yangon


Очікуваний результат:

![](https://drive.google.com/uc?export=view&id=1JdXUmK46WcoXUJYP7NS2SJlr2XjBbpQ8)

8. Для продовження дослідження різниці у прибутках в магазинах давайте зʼясуємо, чи відрізняється маржа ('gross margin percentage') для різних інвойсів в магазині? Перевірте це програмно на даних виходячи з того функціоналу, який ми вже пройшли у лекціях.

Дайте на основі отриманого результату відповідь: "У всіх всіх інвойсах однакова маржа" або "Маржа відрізняється для різних інвойсів".

In [253]:
nunique_gross_margin = df['gross margin percentage'].value_counts()
nunique_gross_margin

gross margin percentage
4.761905    1000
Name: count, dtype: int64

In [254]:
if not len(nunique_gross_margin) > 1:
    print('У всіх всіх інвойсах однакова маржа')

У всіх всіх інвойсах однакова маржа


Очікуваний результат: відповідь щодо маржі в усіх інвойсах на основі отриманих результатів

9. Врахуовуючи висновок в попередньому завданні, дослідимо як продаються в цих містах різні категорії товарів. Виведіть відсотковий розпоіділ кількості інвойсів по категоріям товарів `Product line` в містах `Yangon` та `Mandalay`.

In [255]:
city_df2[city_df2.City == 'Yangon']['Product line'].value_counts(normalize=True)

Product line
Home and lifestyle        0.191176
Electronic accessories    0.176471
Sports and travel         0.173529
Food and beverages        0.170588
Fashion accessories       0.150000
Health and beauty         0.138235
Name: proportion, dtype: float64

In [256]:
city_df2[city_df2.City == 'Mandalay']['Product line'].value_counts(normalize=True)

Product line
Fashion accessories       0.186747
Sports and travel         0.186747
Electronic accessories    0.165663
Health and beauty         0.159639
Food and beverages        0.150602
Home and lifestyle        0.150602
Name: proportion, dtype: float64

Очікуваний результат для кожного з міст:

![](https://drive.google.com/uc?export=view&id=1IbG58YK5KVqC-wCnllF1btgEgfWTbi_O)

![](https://drive.google.com/uc?export=view&id=1aKOkLmlMq1sOh4GbwyaQyjfhNJzdU4Yh)